commands/init.go

```diff
@@ -0,0 +1,102 @@
+package commands
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+// InitCmd creates the necessary files required to deploy an app (server.conf)
+type InitCmd struct {
+	StackName string `optional default:"nodejs-basic" short:"s" help:"Name of stack to deploy. Default is nodejs-basic"`
+}
+
+// Copy and paste of server.conf
+func buildServerConf() []byte {
+	return []byte(
+		`location / {
+	proxy_set_header X-Forwarded-For $http_x_forwarded_for;
+	proxy_set_header X-Forwarded-Proto $http_x_forwarded_proto;
+	proxy_set_header Host $host;
+	include /etc/nginx/section.module/node.conf;
+}
+		
+location ~ "/notnuxt/" {
+	proxy_set_header X-Forwarded-For $http_x_forwarded_for;
+	proxy_set_header X-Forwarded-Proto $http_x_forwarded_proto;
+	proxy_set_header Host $host;
+	proxy_pass http://next-hop;
+}`)
+}
+
+// Run executes the command
+func (c *InitCmd) Run() (err error) {
+	if c.StackName == "nodejs-basic" {
+		fmt.Println("Checking to see if server.conf exists")
+		checkServConf, err := os.Open("server.conf")
+		if err != nil {
+			fmt.Println("WARN: server.conf does not exist. Creating server.conf")
+			f, err := os.Create("server.conf")
+			if err != nil {
+				panic(err)
+			}
+			b := buildServerConf()
+			f.Write(b)
+			defer f.Close()
+		} else {
+			fmt.Println("Validating server.conf")
+			fileinfo, err := checkServConf.Stat()
+			if err != nil {
+				panic(err)
+			}
+			buff := make([]byte, fileinfo.Size())
+			checkServConf.Read(buff)
+			fileString := string(buff)
+			if !strings.Contains(fileString, "location / {") {
+				fmt.Println("WARN: default location unspecified. Edit or delete server.conf and rerun this command")
+			}
+			if !strings.Contains(fileString, "/notnuxt/") {
+				fmt.Println("WARN: notnuxt location unspecified. Edit or delete server.conf and rerun this command")
+			}
+		}
+		defer checkServConf.Close()
+		fmt.Println("Checking to see if package.json exists")
+		checkPkgJSON, err := os.Open("package.json")
+		if err != nil {
+			fmt.Println("WARN: package.json does not exist. Creating package.json")
+			cmd := exec.Command("npm", "init", "-y")
+			cmd.Stdout = os.Stdout
+			cmd.Stderr = os.Stderr
+			err := cmd.Run()
+			if err != nil {
+				fmt.Println("FATAL: There was an error creating package.json. Is node installed?")
+			} else {
+				fmt.Println("package.json created")
+				fmt.Println("WARN: package.json does not have a 'start' script. This is required")
+			}
+		} else {
+			fmt.Println("Validating package.json")
+			fileinfo, err := checkPkgJSON.Stat()
+			if err != nil {
+				panic(err)
+			}
+			buff := make([]byte, fileinfo.Size())
+			checkPkgJSON.Read(buff)
+			fileString := string(buff)
+			if !strings.Contains(fileString, "name") {
```

Comment:
name and version are no longer required, but start is still required yeah.

Reply:
Okay good to know. I can remove that from the PR